frame/conviction-voting/src/lib.rs

```diff
@@ -0,0 +1,625 @@
+		///   Weight is calculated for the maximum number of vote.
+		#[pallet::weight(T::WeightInfo::remove_vote())]
+		pub fn remove_vote(
+			origin: OriginFor<T>,
+			class: Option<ClassOf<T>>,
+			index: PollIndexOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::try_remove_vote(&who, index, class, UnvoteScope::Any)
+		}
+
+		/// Remove a vote for a poll.
+		///
+		/// If the `target` is equal to the signer, then this function is exactly equivalent to
+		/// `remove_vote`. If not equal to the signer, then the vote must have expired,
+		/// either because the poll was cancelled, because the voter lost the poll or
+		/// because the conviction period is over.
+		///
+		/// The dispatch origin of this call must be _Signed_.
+		///
+		/// - `target`: The account of the vote to be removed; this account must have voted for poll
+		///   `index`.
+		/// - `index`: The index of poll of the vote to be removed.
+		/// - `class`: The class of the poll.
+		///
+		/// Weight: `O(R + log R)` where R is the number of polls that `target` has voted on.
+		///   Weight is calculated for the maximum number of vote.
+		#[pallet::weight(T::WeightInfo::remove_other_vote())]
+		pub fn remove_other_vote(
+			origin: OriginFor<T>,
+			target: T::AccountId,
+			class: ClassOf<T>,
+			index: PollIndexOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let scope = if target == who { UnvoteScope::Any } else { UnvoteScope::OnlyExpired };
+			Self::try_remove_vote(&target, index, Some(class), scope)?;
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	/// Actually enact a vote, if legit.
+	fn try_vote(
+		who: &T::AccountId,
+		poll_index: PollIndexOf<T>,
+		vote: AccountVote<BalanceOf<T>>,
+	) -> DispatchResult {
+		ensure!(vote.balance() <= T::Currency::free_balance(who), Error::<T>::InsufficientFunds);
+		T::Polls::try_access_poll(poll_index, |poll_status| {
+			let (tally, class) = poll_status.ensure_ongoing().ok_or(Error::<T>::NotOngoing)?;
+			VotingFor::<T>::try_mutate(who, &class, |voting| {
+				if let Voting::Casting(Casting { ref mut votes, delegations, .. }) = voting {
+					match votes.binary_search_by_key(&poll_index, |i| i.0) {
+						Ok(i) => {
+							// Shouldn't be possible to fail, but we handle it gracefully.
+							tally.remove(votes[i].1).ok_or(ArithmeticError::Underflow)?;
+							if let Some(approve) = votes[i].1.as_standard() {
+								tally.reduce(approve, *delegations);
+							}
+							votes[i].1 = vote;
+						},
+						Err(i) => {
+							ensure!(
+								(votes.len() as u32) < T::MaxVotes::get(),
+								Error::<T>::MaxVotesReached
+							);
+							votes.insert(i, (poll_index, vote));
+						},
+					}
+					// Shouldn't be possible to fail, but we handle it gracefully.
+					tally.add(vote).ok_or(ArithmeticError::Overflow)?;
+					if let Some(approve) = vote.as_standard() {
+						tally.increase(approve, *delegations);
+					}
```

Comment:
What happens when the vote is split here and `as_standard()` returns `None`?

Seems like we are not taking that into account with the tally.

Reply:
yeah it'd be a weird use case and it's not supported.